src/jdk.crypto.cryptoki/share/native/libj2pkcs11/p11_util.c

```diff
@@ -409,11 +409,28 @@ void freeCKMechanismPtr(CK_MECHANISM_PTR mechPtr) {
                  case CKM_CAMELLIA_CTR:
                      // params do not contain pointers
                      break;
+                 case CKM_PKCS5_PBKD2:
+                     // get the versioned structure from behind memory
+                     TRACE0(((VersionedPbkd2ParamsPtr)tmp)->version == PARAMS ?
+                             "[ CK_PKCS5_PBKD2_PARAMS ]\n" :
+                             "[ CK_PKCS5_PBKD2_PARAMS2 ]\n");
+                     FREE_VERSIONED_PBKD2_MEMBERS((VersionedPbkd2ParamsPtr)tmp);
+                     break;
+                 case CKM_PBA_SHA1_WITH_SHA1_HMAC:
+                 case CKM_NSS_PKCS12_PBE_SHA224_HMAC_KEY_GEN:
+                 case CKM_NSS_PKCS12_PBE_SHA256_HMAC_KEY_GEN:
+                 case CKM_NSS_PKCS12_PBE_SHA384_HMAC_KEY_GEN:
+                 case CKM_NSS_PKCS12_PBE_SHA512_HMAC_KEY_GEN:
```

Comment:
How about CKM_PBE_SHA1_DES3_EDE_CBC and CKM_PBE_SHA1_DES2_EDE_CBC?

Reply:
Good point. While CKM_PBE_SHA1_DES3_EDE_CBC and CKM_PBE_SHA1_DES2_EDE_CBC mechanisms were never supported, there was dead code in jMechParamToCKMechParamPtrSlow (p11_convert.c) for them before this enhancement proposal. In fact, CK_PBE_PARAMS (Java and C classes) existed without any uses, probably added for the sake of comprehensiveness. This code had issues that we had to fix when actually using it. It's not our intention to include legacy derivation mechanisms, such as PBEWithSHA1AndDESede, at this time in the scope. I'll propose to remove all references to CKM_PBE_SHA1_DES3_EDE_CBC and CKM_PBE_SHA1_DES2_EDE_CBC in jMechParamToCKMechParamPtrSlow so it don't lead to confusion.